nyxx_commander/lib/src/utils.dart

```diff
@@ -1,11 +1,10 @@
 part of nyxx_commander;
 
-// TODO: FIX. I think that is just awful but is does its job
-extension _CommandMatcher on Iterable<CommandEntity> {
-  CommandEntity? _findMatchingCommand(Iterable<String> messageParts) {
-    for (final entity in this) {
-      if (entity is CommandGroup && entity.name == "") {
-        final e = entity._commandEntities._findMatchingCommand(messageParts);
+class  _CommandMatcher {
+  static CommandEntity? _findMatchingCommand(Iterable<String> messageParts, Iterable<CommandEntity> commands) {
```

Comment:
Does this need documentations? Possibly for contributors? same for class line above.